lib/ansible/modules/cloud/amazon/iam_cert_facts.py

```diff
@@ -0,0 +1,158 @@
+    sample: {"HTTPStatusCode": 200, "RequestId": "9f6d-671298e72d8b"}
+
+ServerCertificateMetadataList:
+    description: Metadata list.
+    returned: always
+    type: dict
+    sample: http://boto3.readthedocs.io/en/latest/reference/services/iam.html#IAM.Client.list_server_certificates
+'''
+
+try:
+    import boto3
+    import botocore
+
+    from botocore.exceptions import ClientError, NoAuthHandlerFound
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
+
+
+def get_server_certificate(client, module, server_certificate_name):
+    """ Retrieves information about the specified server certificate stored in IAM.
+    """
+    try:
+        response = client.get_server_certificate(
+            ServerCertificateName=server_certificate_name)
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'NoSuchEntity':
+            return e.response['Error']
+        else:
+            raise
+    except Exception as e:
+        module.fail_json(msg="Unable to get server certificate :: %s" % str(e))
+    return response
+
+
+def list_server_certificates(client, module, path_prefix):
+    """ Lists the server certificates stored in IAM that have the specified path prefix.
+        If none exist, the action returns an empty list.
+    """
+    try:
+        if path_prefix:
+            response = client.list_server_certificates(PathPrefix=path_prefix)
+        else:
+            response = client.list_server_certificates()
+    except Exception as e:
+        module.fail_json(msg="Unable to get server certificates :: %s" % str(e))
+    return response
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        certificate_name=dict(type='str', default=None),
+        path_prefix=dict(type='str', default=None),
+    ))
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        mutually_exclusive=[
+            ['certificate_name', 'path_prefix'],
+        ],
+        supports_check_mode=False,
+    )
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto/boto3 is required.')
+
+    try:
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+        iam_cert = boto3_conn(module, conn_type='client', resource='iam',
+                              region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except NoAuthHandlerFound as e:
+        module.fail_json(msg="Can't authorize connection - %s " % str(e))
+
+    certificate_name = module.params['certificate_name']
+    path_prefix = module.params['path_prefix']
+
+    if certificate_name:
+        results = get_server_certificate(
+            client=iam_cert, module=module, server_certificate_name=certificate_name)
+    else:
+        results = list_server_certificates(client=iam_cert, module=module, path_prefix=path_prefix)
```

Comment:
Please use the camel_dict_to_snake_dict to fix the camelcase in the boto3 responses here.